lib/scholar/metrics.ex

```diff
@@ -646,6 +646,65 @@ defmodule Scholar.Metrics do
     (diff * diff) |> Nx.mean()
   end
 
+  @doc ~S"""
+  Calculates the mean square logarithmic error of predictions
+  with respect to targets.
+
+  $$MSLE = \frac{\sum_{i=1}^{n} (\log(\hat{y_i} + 1) - \log(y_i + 1))^2}{n}$$
+
+  ## Examples
+
+      iex> y_true = Nx.tensor([[0.0, 1.0], [0.0, 0.0]])
+      iex> y_pred = Nx.tensor([[1.0, 1.0], [1.0, 0.0]])
+      iex> Scholar.Metrics.mean_square_log_error(y_true, y_pred)
+      #Nx.Tensor<
+        f32
+        0.24022650718688965
+      >
+  """
+  defn mean_square_log_error(y_true, y_pred) do
+    assert_same_shape!(y_true, y_pred)
+
+    (Nx.log(y_true + 1) - Nx.log(y_pred + 1))
+    |> Nx.pow(2)
+    |> Nx.mean()
+  end
+
+  @doc ~S"""
+  Calculates the mean absolute percentage error of predictions
+  with respect to targets. If `y_true` values are equal or close
+  to zero, it returns an arbitrarily large value.
+
+  $$MAPE = \frac{\sum_{i=1}^{n} \frac{|\hat{y_i} - y_i|}{max(\epsilon, \hat{y_i})}}{n}$$
+
+  By default $\epsilon = 1^{-10}$.
+
+  ## Examples
+
+      iex> y_true = Nx.tensor([3, -0.5, 2, 7])
+      iex> y_pred = Nx.tensor([2.5, 0.0, 2, 8])
+      iex> Scholar.Metrics.mean_absolute_percentage_error(y_true, y_pred)
+      #Nx.Tensor<
+        f32
+        0.3273809552192688
+      >
+
+      iex> y_true = Nx.tensor([1.0, 0.0, 2.4, 7.0])
+      iex> y_pred = Nx.tensor([1.2, 0.1, 2.4, 8.0])
+      iex> Scholar.Metrics.mean_absolute_percentage_error(y_true, y_pred)
+      #Nx.Tensor<
+        f32
+        2.5e8
+      >
+  """
+  defn mean_absolute_percentage_error(y_true, y_pred) do
+    assert_same_shape!(y_true, y_pred)
+    eps = Nx.tensor(1.0e-10)
```

Comment:
```suggestion
    eps = Nx.Constants.epsilon({:f, 64})
```

Reply:
Should the epsilon be based on the input type?

Reply:
I don't think so, because eps(:f32) will be almost always too big (~1.0e-7)

Reply:
If the role of an eps is to replace 0s in division then we almost always will use `:f64`

Reply:
On the other hand, if we generate noise in the algorithm and the standard deviation depends on eps, then eps should be dependent on the type

Reply:
The problem is that `Nx.Constants.epsilon(:f64)` forces output to be `:f64` but ideally we would preserve type if we could (`:f32`), but use the value of `Nx.Constants.epsilon(:f64)`

Reply:
Doesn’t it mean we should have a lower value for epsilon on f32 then?

Reply:
The epsilon value we have is well-defined and is the same that numpy provides.

I believe it's the smallest value where 1+eps returns an accurate result.

Reply:
Well, to be honest, it's not important which convention we eventually choose.  The result if `eps` is max in the denominator, then we end up with a huge result which can only be interpreted as a nearly zero value somewhere in `y_true` (See second doctest). So I'm ok with either version. Code for type dependant version would be `eps = to_float_type(Nx.Type.merge(Nx.type(x), Nx.type(y))`